### PR TITLE
Ignore citations that don't correspond to search results

### PR DESCRIPTION
**Pull request by ofermend**
A user reported randomly encountering this error after issuing a search request:

image

This is due to the text of the corpus itself containing citations, e.g. a text segment like "here is some text [2344]."

The user is using prompt v1.2.0 which includes these citations in the summary. The long-term solution is to use an improved prompt that will exclude these citations, but this PR introduces a short-term solution. This solution ignores citations that don't correspond to search results. Note that this isn't a fool-proof solution because the corpus text could still contain citations that collide with actual search results. Those citations won't result in the above error, but they will be erroneously linked to the actual search results.

### CODE REVIEW

File: src/utils/deserializeSearchResponse.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ const convertMetadataToObject = (metadata: DocMetadata[]) => {
 const parseMetadata = (rawMetadata: DocMetadata[]) => {
   const metadata = convertMetadataToObject(rawMetadata);
   return {
-    source: metadata.source as string,
+    source: metadata.source,
```
Comment (cjcenizal) [inline src/utils/deserializeSearchResponse.ts:15]: Why is this needed?

**Review by cjcenizal — COMMENTED** [review]
Thanks for making this change @ofermend! I'm having trouble understanding the original problem which makes it tough for me to review this change. I left one comment, but a couple other things would help me understand this better:

1. A step-by-step explanation of how this bug occurs and what it manifests as. Steps to repro would be ideal but I understand if that's tricky to do, e.g. if this is non-deterministic.
2. A new test case added to https://github.com/vectara/vectara-answer/blob/d710be1a8b318ddb3d2e6e53ce70b21ece792390/src/ui/utils/citations/applyCitationOrder.test.ts that will serve as documentation of this edge case, as well as guard against regression in the future. Happy to help with this!

File: src/utils/deserializeSearchResponse.ts (hunk 1 above)
Comment (ofermend) [reply src/utils/deserializeSearchResponse.ts:15]: It's not. Old. Removing.

**Review by ofermend — COMMENTED** [review]
LGTM
